### PR TITLE
Bug 1141289 - Replace local imports with global import

### PR DESCRIPTION
**Pull request by tareqak**

### CODE REVIEW

File: firefox_puppeteer/ui/about_window/window.py [hunk 1]
```diff
@@ -133,3 +133,5 @@ def wait_for_update_applied(self, timeout=TIMEOUT_UPDATE_APPLY):
         Wait(self.marionette, timeout=timeout).until(
             lambda _: 'applied' in self._software_update.active_update.state,
             message='Update has been applied.')
+
+Windows.register_window(AboutWindow.window_type, AboutWindow)
```
Comment (whimboo) [inline firefox_puppeteer/ui/about_window/window.py:137]: Nice approach in registering window classes. That makes total sense.

File: firefox_puppeteer/ui/browser/window.py [hunk 2]
```diff
@@ -8,12 +8,16 @@
...
+from firefox_puppeteer.ui.windows import BaseWindow, Windows
 import firefox_puppeteer.errors as errors
 
 from firefox_puppeteer.api.l10n import L10n
 from firefox_puppeteer.decorators import use_class_as_property
 from firefox_puppeteer.api.prefs import Preferences
+from ..pageinfo.window import PageInfoWindow
+from .toolbars import NavBar
+from .tabbar import TabBar
+from ..about_window.window import AboutWindow
```
Comment (whimboo) [inline firefox_puppeteer/ui/browser/window.py]: Lets follow the absolute module path pattern like `from firefox_puppeteer.ui.pageinfo.window import PageInfoWindow`.

File: firefox_puppeteer/ui/windows.py [hunk 3]
```diff
@@ -186,6 +181,16 @@ def switch_to(self, target):
 
         return self.create_window_instance(target_handle)
 
+    @classmethod
+    def register_window(cls, window_type, window_constructor):
+        """Registers a chrome window with this class so that this class may in
+        turn create the appropriate window instance later on.
+
+        :param window_type: The type of window.
+        :param window_constructor: The constructor of the window
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: nit: It's the class we pass in here and not the constructor. So call it `window_class`.

File: firefox_puppeteer/ui_base_lib.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+from firefox_puppeteer.base import BaseLib
+
+
+class UIBaseLib(BaseLib):
+    """A base class for all UI element wrapper classes inside a chrome window."""
+
+    def __init__(self, marionette_getter, window, element):
+        # importing globally doesn't work
+        from .ui.windows import BaseWindow
```
Comment (whimboo) [inline firefox_puppeteer/ui_base_lib.py]: This local import we cannot move to a global one?

File: firefox_puppeteer/base.py [hunk 5]
```diff
@@ -23,34 +21,3 @@ def marionette(self):
 
     def get_marionette(self):
         return self.marionette
-
-
-class UIBaseLib(BaseLib):
-    """A base class for all UI element wrapper classes inside a chrome window."""
```
Comment (whimboo) [inline firefox_puppeteer/base.py:29]: So this separation in its own module is necessary?

File: firefox_puppeteer/ui_base_lib.py (hunk 4 above)
Comment (tareqak) [inline firefox_puppeteer/ui_base_lib.py]: I missed that one. I'll move it up.

File: firefox_puppeteer/base.py (hunk 5 above)
Comment (tareqak) [inline firefox_puppeteer/base.py:29]: Yes. BaseLib and UIBaseLib being in the same file was causing one of the circular dependencies.

File: firefox_puppeteer/ui/windows.py (hunk 3 above)
Comment (tareqak) [inline firefox_puppeteer/ui/windows.py]: I'll update the pull request with the name change.

File: firefox_puppeteer/ui/browser/window.py (hunk 2 above)
Comment (tareqak) [inline firefox_puppeteer/ui/browser/window.py]: I'll make this change, and update all the other files while I am at it.

**Comment by tareqak** [comment]
It seems like the travis couldn't find the build to test my commit against.

**Comment by whimboo** [comment]
Yes, Releng busted our tests when they moved the Firefox builds from our FTP to S3. I meanwhile fixed all dependencies so tests should pass now. I will retrigger them.

**Comment by whimboo** [comment]
Just two more nits to get fixed. All the rest looks wonderful. Would you mind to also squash all the commits into a single one and give it a proper comment? We normally use "Bug XYZ - summary. r=whimboo". Thanks.

**Comment by tareqak** [comment]
I made the changes. I have been making the changes and committing them by doing:
-< make changes >-
-< add changes >-
git commit --amend
git push --force

At the top, it shows that there is only one commit. I guess Github could make it a little more obvious that the commit itself has changed.

**Comment by whimboo** [comment]
Excellent! All looks perfect on this PR now so that I can get it landed in a couple of minutes.

**Comment by whimboo** [comment]
I merged the PR into mozilla-central as https://github.com/mozilla/firefox-ui-tests/commit/acd30465aa9d1372edd2a402cd801ad20319ed4d.

**Pull request closed without merging**